js/MarkupSubmissionsBatchConversion.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file js/MarkupSubmissionsBatchConversion.js
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @package plugins.generic.markup
+ * @class MarkupSubmissionsBatchConversion
+ *
+ * @brief Batch conversion client side handler
+ */
+(function($) {
+
+	/** @type {Object} */
+	$.pkp.plugins.markup =
+			$.pkp.plugins.markup ||
+			{ js: { } };
+
+	/**
+	 * @constructor
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion = function() {
+
+		// batch conversion button click handler
+		$('div.pkp_form button#startConversionBtn').click(function() {
+			$(this).attr('disabled','disabled').hide();
+			$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.processFile();
+		});
+	};
+
+	/**
+	 * A Number, representing the ID value of the timer that is set.
+	 * @private
+	 * @type {int}
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.timer = null;
+
+	/**
+	 * A jQuery element.
+	 * @private
+	 * @type {jQueryObject}
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.$listElement = null;
+
+	/**
+	 * Determine which file to convert
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.processFile = 
+		function() {
+			this.conversionStatus();
+			var $next = $('select[class=submission-file]:not(".processed")').first();
+			if ($next.length) {
+				this.convert($next);
+			}
+	}
+
+	/**
+	 * Display numbers of submissions processed and to be processed
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.conversionStatus = 
+		function() {
+			$('div#conversion-status').show();
+			var processedCount = $('select[class="submission-file processed"]').size();
+			var totalCount = $('ul#submissionListConfirmation select').size();
+			$('span#conversion-status-processed').text(processedCount);
+			$('span#conversion-status-total').text(totalCount);
+	}
+
+	/**
+	 * Trigger file conversion
+	 * 
+	 * @param {jQueryObject} $element input checkbox representing submission file
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.convert = 
+		function($element) {
+			$element.addClass('processed');
+			this.$listElement = $element.closest('li');
+			this.$listElement.addClass('batch-processing');
+			var params = {
+				'submissionId': $element.data('submission-id'),
+				'stage': $element.data('stage'),
+				'fileId': parseInt($element.val()),
+			};
+
+			if (params.fileId != -1) {
+				var postUrl = $('input#conversionTriggerUrl').val() +
+					'?submissionId='+params.submissionId+

```

Comment:
If `path_info_disabled` is turned on, there will probably already be a `?` in the URL. The URLs should be fully built in PHP code, and if you need to make new ones JS-side, use placeholders on the PHP side and then do string replacement on the JS side.

Reply:
Build the URLs in PHP code would mean to do it for each submission. So if we have 100 submissions, we will have 100 URLs in the HTML.

How about I check if the url already contains `?` ? if so I append only `&`  otherwise I append `?` and `&`.

Reply:
Use placeholders on the PHP side. For example, build the URL with parameters like

    array('submissionId' => 'SUBMISSION_ID_SLUG')

...etc, and then on the JS side, use...

   var theRealUrl = this.templateUrl.replace('SUBMISSION_ID_SLUG', submissionId);

Reply:
my concern was instead of having to search and replace three times (for `submissionId`, `stage` and `fileId`), what about we look for `?` in the url and append `?` or `&`? Let me know your preference.

Reply:
It would just be...

    myUrl = someString.replace('SUBMISSION_ID_SLUG', submissionId)
        .replace('ANOTHER_SLUG', anotherId)
        .replace('YET_ANOTHER_SLUG', yetAnotherId);

...which I think is cleaner than looking for `?` or `&` and splitting the URL building, which is currently entirely on the PHP side, partially into the Javascript.